builder/make-config-files.nix

```diff
@@ -75,6 +75,9 @@ in { identifier, component, fullName, flags ? {} }:
 
     echo ${lib.concatStringsSep " " (lib.mapAttrsToList (fname: val: "--flags=${lib.optionalString (!val) "-" + fname}") flags)} >> $out/configure-flags
 
+    # Provide a cabal config without remote package repositories
+    echo "write-ghc-environment-files: never" >> $out/cabal.config
```

Comment:
This comment and line don't obviously belong together - is the idea that we put *something* minimal in the config file?

Reply:
I'm also confused about why we didn't need this before.

Reply:
Yes the comment is a bit misleading. I decided to add this because `CABAL_CONFIG` will override the user's `cabal.config`, and `.ghc.environment.*` is not needed at all under shellFor.